File: Week_Two/Problem_Set/readability.c
#include<stdio.h>
#include<cs50.h>
#include<ctype.h>
#include<string.h>
#include<math.h>

int countLetters(string usersString);

int main(void)
{
    //Get users input
    string userText = get_string("Text: ");
    int userTextLength = countLetters(userText);
    //printf("Grade %i\n", );

    return 0;
}

//Count overall length of users input
int countLetters(string usersString)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    for(int i = 0, n = strlen(usersString); i < n; i++)
    {
        if(isalpha(usersString[i]))
        {
            letters++;
        }
        else if(isspace(usersString[i]))
        {
            words++;
        }
        else if(usersString[i] == '!' || usersString[i] == '?' || usersString[i] == '.')
        {
            sentences++;
        }
    }
    printf("letters: %i\n", letters);
    printf("spaces: %i\n", words);
    printf("sentences: %i\n", sentences);

    float L = (((float) letters / words) * 100);
    float S = (((float) sentences / words) * 100);
    printf("test: %f\n", L);
    printf("test: %f\n", S);
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    printf("index: %i", index);


    return letters;
}





/*
index = 0.0588 * L - 0.296 * S - 15.8
Here, L is the average number of letters per 100 words in the text, and S is the average number of sentences per 100 words in the text.



Example = Congratulations! Today is your day. You're off to Great Places! You're off and away!
The text the user inputted has 65 letters, 4 sentences, and 14 words.


65 letters per 14 words is an average of about 464.29 letters per 100 words.
And 4 sentences per 14 words is an average of about 28.57 sentences per 100 words.
Plugged into the Coleman-Liau formula, and rounded to the nearest whole number, we get an answer of 3: so this passage is at a third grade reading level.


Your program should print as output "Grade X" where X is the grade level computed by the Coleman-Liau formula, rounded to the nearest integer.
If the resulting index number is 16 or higher (equivalent to or greater than a senior undergraduate reading level),
your program should output "Grade 16+" instead of giving the exact index number.
If the index number is less than 1, your program should output "Before Grade 1".


*/

File: Week_6/Pset/readability/readability.py
"""
    Re-implementation of readability.c in the python language.
    A program that calculates the reading level of a users supllied text,
    according to grade level using the Coleman-Liau formula.
"""

from cs50 import get_string


def main():

    # Get text to be graded
    user_text = get_string("Text: ")

    # Send to formula
    grade = grade_level(user_text)

    # Print grade level
    print_grade(grade)

    return


def grade_level(n):

    # Letter, word, sentence count
    letter, word, sentence = 0, 1, 0

    # Loop through text
    for i in n:

        # Increment letters
        if i.isalpha():
            letter += 1

        # Count spaces as start of new words
        if i.isspace():
            word += 1

        # Use these 3 symbols as end of sentence identifiers
        if i in ['?', '.', '!']:
            sentence += 1

    # Apply Coleman-Liau formula
    L = (letter / word) * 100
    S = (sentence / word) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    return index


def print_grade(n):

    # Handle Kindergarten and below
    if n < 1:
        print("Before Grade 1")

    # Handle reading levels above 16
    elif n > 16:
        print("Grade 16+")

    # Print grade
    else:
        print(f"Grade {n}")

    return


main()
